string grammar
string[] vaild_gram_id
---
bool success
